### ISSUE TITLE
Deprecate the plain text authentication method from AWS and Azure

### ISSUE THREAD

**Issue by CarlosRS9** [wazuh/wazuh#14508]
| Affected integration |
|---|
| AWS and Azure |

## Description

In https://github.com/wazuh/wazuh/issues/13660 we have reviewed the different authentication options and we have concluded that we should deprecate and remove the option that allows the user to store their credentials as plain text in the ossec.conf. We will indicate them that it is necessary to use credential files instead.

We will follow the following steps for its removal:

1. In `4.4` we will deprecate the user and password parameters, updating the ossec.conf parser to show a warning message if one of them are being used. The idea is to warn the user that for the moment they are still accepted, but in the next major they will be completely removed.
2. In `5.0` we will remove these options. A configuration containing one of those will result in an error and the module will not run.

Having this in mind, we need to apply the following changes for `4.4`:

1. Update the ossec.conf parser to detect these parameters and show a warning message.
2. Update `-h` option of the python module to show that these parameters are now deprecated.
3. Update the python module to show a warning if the user try to manually run it using these parameters.
4. Update the Wazuh documentation to acomodate these changes.

## Tasks
- [x] Test in a manager.
- [x] Test in an agent.
- [x] Unit tests without failures. Updated if there are any relevant changes.
- [x] Integration tests without failures. Updated if there are any relevant changes.
- [x] Update the documentation if necessary.
- [x] Add entry to the changelog if necessary.

**Comment by CarlosRS9** [comment]
## Ossec.conf parser changes
We have already implemented the necessary changes in our `ossec.conf` parser to detect the deprecated tags and warn the user with a message.

When a deprecated tag, such as `access_key` is detected the following messages are shown:
```
2022/08/08 12:51:36 wazuh-modulesd: WARNING: Deprecated tag <access_key> found at module 'aws-s3'. This tag was deprecated in 4.4. Please, use a different authentication method.
2022/08/08 12:51:36 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/amazon/services/prerequisites/credentials.html for more information.
```
These messages are shown as part of the Wazuh service output instead of the ossec.log, which means they will be visible when the user attempts to start or restart the Wazuh service.

Here is an example of a real configuration using AWS buckets, AWS services, Azure Graph, Azure Log Analytics y Azure Storage:

<details><summary>Configuration used</summary>

	<wodle name="aws-s3">
	    <disabled>no</disabled>
	    <remove_from_bucket>no</remove_from_bucket>
	    <interval>10m</interval>
	    <run_on_start>yes</run_on_start>
	    <skip_on_error>no</skip_on_error>

	    <bucket type="cloudtrail">
		<access_key>xxxxxx</access_key>
		<secret_key>xxxxxx</secret_key>
		<name>wazuh-aws-wodle-cloudtrail</name>
		<only_logs_after>2021-NOV-21</only_logs_after>
	    </bucket>

	    <bucket type="custom">
		<access_key>xxxxxx</access_key>
		<secret_key>xxxxxx</secret_key>
		<name>wazuh-aws-wodle-macie</name>
		<only_logs_after>2021-NOV-10</only_logs_after>
	    </bucket>

	    <service type="cloudwatchlogs">
		<access_key>xxxxxx</access_key>
		<secret_key>xxxxxx</secret_key>
		<aws_log_groups>4_3_test</aws_log_groups>
		<only_logs_after>2021-AUG-20</only_logs_after>
		<regions>us-east-1</regions>
	    </service>
	</wodle>



	<wodle name="azure-logs">
	  <disabled>no</disabled>
	  <interval>10m</interval>
	  <run_on_start>yes</run_on_start>

	  <graph>
	    <application_id>8b7...c14</application_id>
	    <application_key>w22...91x</application_key>

	    <tenantdomain>wazuh.onmicrosoft.com</tenantdomain>
	    <request>
		<query>auditLogs/directoryAudits</query>
	    </request>
	  </graph>

	  <log_analytics>
	    <application_id>8b7...c14</application_id>
	    <application_key>w22...91x</application_key>
	    <tenantdomain>wazuh.onmicrosoft.com</tenantdomain>

	    <request>
		<tag>azure-diagnostics</tag>
		<query>AzureDiagnostics</query>
		<workspace>xxxxxxxxxxxxx</workspace>
	    </request>

	  </log_analytics>
	  
	  <storage>
	      <account_name>exampleaccountname</account_name>
	      <account_key>w22...91x</account_key>
	      <tag>azure-activity</tag>

	      <container name="frameworktestcontainer">
		  <content_type>json_inline</content_type>
		  <time_offset>50d</time_offset>
	      </container>
	  </storage>
	</wodle>
</details>

<details><summary>Service output</summary>

	service wazuh-manager restart

	wazuh-clusterd not running...
	Killing wazuh-modulesd...
	Killing wazuh-monitord...
	Killing wazuh-logcollector...
	Killing wazuh-remoted...
	Killing wazuh-syscheckd...
	Killing wazuh-analysisd...
	wazuh-maild not running...
	Killing wazuh-execd...
	Killing wazuh-db...
	wazuh-authd not running...
	wazuh-agentlessd not running...
	wazuh-integratord not running...
	wazuh-dbd not running...
	wazuh-csyslogd not running...
	Killing wazuh-apid...
	Wazuh v4.5.0 Stopped
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Deprecated tag <access_key> found at module 'aws-s3'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/amazon/services/prerequisites/credentials.html for more information
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Deprecated tag <secret_key> found at module 'aws-s3'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/amazon/services/prerequisites/credentials.html for more information
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Deprecated tag <access_key> found at module 'aws-s3'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/amazon/services/prerequisites/credentials.html for more information
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Deprecated tag <secret_key> found at module 'aws-s3'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/amazon/services/prerequisites/credentials.html for more information
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Deprecated tag <application_id> found at module 'azure-logs'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/azure/activity-services/prerequisites/credentials.html for more information
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Deprecated tag <application_key> found at module 'azure-logs'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/azure/activity-services/prerequisites/credentials.html for more information
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Deprecated tag <application_id> found at module 'azure-logs'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/azure/activity-services/prerequisites/credentials.html for more information
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Deprecated tag <application_key> found at module 'azure-logs'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/azure/activity-services/prerequisites/credentials.html for more information
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Deprecated tag <account_name> found at module 'azure-logs'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/azure/activity-services/prerequisites/credentials.html for more information
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Deprecated tag <account_key> found at module 'azure-logs'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:12 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/azure/activity-services/prerequisites/credentials.html for more information
	Starting Wazuh v4.5.0...
	Started wazuh-apid...
	Started wazuh-csyslogd...
	Started wazuh-dbd...
	2022/08/08 13:02:14 wazuh-integratord: INFO: Remote integrations not configured. Clean exit.
	Started wazuh-integratord...
	Started wazuh-agentlessd...
	Started wazuh-db...
	Started wazuh-execd...
	Started wazuh-analysisd...
	2022/08/08 13:02:15 wazuh-syscheckd: INFO: (6678): No directory provided for syscheck to monitor.
	2022/08/08 13:02:15 wazuh-syscheckd: INFO: (6001): File integrity monitoring disabled.
	2022/08/08 13:02:15 rootcheck: INFO: Rootcheck disabled.
	Started wazuh-syscheckd...
	Started wazuh-remoted...
	Started wazuh-logcollector...
	Started wazuh-monitord...
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Deprecated tag <access_key> found at module 'aws-s3'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/amazon/services/prerequisites/credentials.html for more information
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Deprecated tag <secret_key> found at module 'aws-s3'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/amazon/services/prerequisites/credentials.html for more information
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Deprecated tag <access_key> found at module 'aws-s3'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/amazon/services/prerequisites/credentials.html for more information
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Deprecated tag <secret_key> found at module 'aws-s3'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/amazon/services/prerequisites/credentials.html for more information
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Deprecated tag <application_id> found at module 'azure-logs'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/azure/activity-services/prerequisites/credentials.html for more information
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Deprecated tag <application_key> found at module 'azure-logs'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/azure/activity-services/prerequisites/credentials.html for more information
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Deprecated tag <application_id> found at module 'azure-logs'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/azure/activity-services/prerequisites/credentials.html for more information
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Deprecated tag <application_key> found at module 'azure-logs'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/azure/activity-services/prerequisites/credentials.html for more information
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Deprecated tag <account_name> found at module 'azure-logs'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/azure/activity-services/prerequisites/credentials.html for more information
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Deprecated tag <account_key> found at module 'azure-logs'. This tag was deprecated in 4.4. Please, use a different authentication method.
	2022/08/08 13:02:16 wazuh-modulesd: WARNING: Check https://documentation.wazuh.com/current/azure/activity-services/prerequisites/credentials.html for more information
	Started wazuh-modulesd...
	Completed.
</details>

Due to the way our Wazuh service is implemented the messages appear two time when starting or restarting the service.

**Comment by CarlosRS9** [comment]
## Python module changes

We have updated the descriptions of the parameters that appear in the Azure and AWS module help output when the `-h` parameter is used, indicating that they have been deprecated.

We have updated the Azure and AWS modules to display deprecation messages when attempting to use the deprecated parameters. However, these messages does not appear in the `ossec.log` unless the user enables debug mode at level 1 or 2, due to the way the capturing of these messages by the core side is implemented. These warning messages are ignored unless debug is enabled.

**Comment by CarlosRS9** [comment]
# Unit test results

```
======================================================== test session starts ========================================================
platform linux -- Python 3.10.5, pytest-6.2.5, py-1.11.0, pluggy-0.13.1
rootdir: /home/carlos/GIT/wazuh/wodles
plugins: cov-3.0.0
collected 267 items                                                                                                                 

aws/tests/test_aws.py ......................................                                                                  [ 14%]
azure/tests/test_azure.py ................................................................................................... [ 51%]
................                                                                                                              [ 57%]
azure/tests/test_orm.py .......................                                                                               [ 65%]
docker-listener/tests/test_docker_listener.py ...................                                                             [ 73%]
gcloud/tests/test_bucket.py .................................                                                                 [ 85%]
gcloud/tests/test_gcloud.py .........                                                                                         [ 88%]
gcloud/tests/test_integration.py ........                                                                                     [ 91%]
gcloud/tests/test_subscriber.py ..............                                                                                [ 97%]
gcloud/tests/test_tools.py ........                                                                                           [100%]

================================================= 267 passed, 84 warnings in 2.35s ==================================================
```

It was necessary to update one Azure unit test to accomodate the new deprecated credentials messages.

**Comment by CarlosRS9** [comment]
## Updating documentation

https://github.com/wazuh/wazuh-documentation/pull/5510 adds a disclaimer that warns the user that this authentication method is deprecated and will be removed:

![image](https://user-images.githubusercontent.com/54286452/184090432-95922668-259c-4640-92e6-c7075e89f139.png)

This PR is going to be merged in `4.3.7`. If in the future we determine that it is necessary to further update the documentation to add more information about this deprecation process we will open a new PR once the `4.3.7` changes have been applied to the `4.4` branch of Wazuh documentation.
